Build failed for me with `nix build '.#pkgsCross.aarch64-multiplatform.tdlib'`.

<details><summary>log</summary>

       last 25 log lines:
       > [  6%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/UdpSocketFd.cpp.o
       > [  6%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/uname.cpp.o
       > [  6%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/user.cpp.o
       > [  6%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/wstring_convert.cpp.o
       > [  7%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/Epoll.cpp.o
       > [  7%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/EventFdBsd.cpp.o
       > [  7%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/EventFdLinux.cpp.o
       > [  7%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/EventFdWindows.cpp.o
       > [  7%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/Iocp.cpp.o
       > [  8%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/KQueue.cpp.o
       > [  8%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/NativeFd.cpp.o
       > [  8%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/Poll.cpp.o
       > [  8%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/Select.cpp.o
       > [  9%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/ThreadIdGuard.cpp.o
       > [  9%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/ThreadPthread.cpp.o
       > make[2]: *** No rule to make target '/build/source/tdutils/generate/auto/mime_type_to_extension.cpp', needed by 'tdutils/CMakeFiles/tdutils.dir/generate/auto/mime_type_to_extension.cpp.o'.  Stop.
       > make[2]: *** Waiting for unfinished jobs....
       > [  9%] Building CXX object tdutils/CMakeFiles/tdutils.dir/td/utils/port/detail/WineventPoll.cpp.o
       > make[1]: *** [CMakeFiles/Makefile2:576: tdutils/CMakeFiles/tdutils.dir/all] Error 2
       > make[1]: *** Waiting for unfinished jobs....
       > [  9%] Linking CXX static library libtdtl.a
       > [  9%] Built target tdtl
       > [  9%] Linking C static library libtdsqlite.a
       > [  9%] Built target tdsqlite
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/7k1nsif1iqmmsmn5dwp5ny0idrawvlsw-tdlib-aarch64-unknown-linux-gnu-1.8.42.drv'.
</details>

--- a/pkgs/by-name/td/tdlib/package.nix
+++ b/pkgs/by-name/td/tdlib/package.nix
@@ -10,6 +10,7 @@
   writeShellApplication,
   common-updater-scripts,
   jq,
+  buildPackages,
 }:
 
 let
@@ -49,12 +50,28 @@ stdenv.mkDerivation {
   };
 
   buildInputs = [
-    gperf
     openssl
     readline
     zlib
   ];
-  nativeBuildInputs = [ cmake ];
+
+  nativeBuildInputs = [
+    cmake
+    gperf
+  ];
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+
+  preConfigure = lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    cmake -B native-build \
+      -DCMAKE_C_COMPILER=$CC_FOR_BUILD \
+      -DCMAKE_CXX_COMPILER=$CXX_FOR_BUILD \
+      -DCMAKE_AR=$(command -v $AR_FOR_BUILD) \
+      -DCMAKE_RANLIB=$(command -v $RANLIB_FOR_BUILD) \
+      -DCMAKE_STRIP=$(command -v $STRIP_FOR_BUILD) \
+      -DTD_GENERATE_SOURCE_FILES=ON .
+    cmake --build native-build -j $NIX_BUILD_CORES
+  '';
 
   # https://github.com/tdlib/td/issues/1974
   postPatch =

